tests/models/gemma/test_modeling_gemma.py

```diff
@@ -570,16 +571,24 @@ def test_model_2b_fp16_static_cache(self):
     @require_read_token
     def test_model_2b_bf16(self):
         model_id = "google/gemma-2b"
-        EXPECTED_TEXTS = {
-            7: [
-                "Hello I am doing a project on the 1990s and I need to know what the most popular music",
-                "Hi today I am going to share with you a very easy and simple recipe of <strong><em>Khichdi",
-            ],
-            8: [
-                "Hello I am doing a project on the 1990s and I need to know what the most popular music",
-                "Hi today I am going to share with you a very easy and simple recipe of <strong><em>Kaju Kat",
-            ],
-        }
+        if IS_ROCM_SYSTEM:
```

Comment:
is this logic needed ? you could just add the key `9` in the `EXPECTED_TEXTS` dict here no?

Reply:
@younesbelkada  @ydshieh  I have been comparing the generate output on H100 and MI300 (both with 9 device capability) and have seen deviations in the generated text. After discussing this with AMD engineers, this is expected as long as output is sensible. 

> The deviation can happen because different hardware may process differently, and if there are non-linearities, minor deviations tend to get amplified.

These deviations might not manifest across all models or prompts (as seen in this particular test). But there are few tests in this PR where the output can be different and thus handled separately for ROCM. 

We have a few options to address this:
1. Since, we do not have H100 tests (?) I could merge the dict, and separate them later if necessary.
2. Have if/else statements only for tests where the generated output is different.
3. Distinguish the ROCM `EXPECTED` output for each tests as done currently in the PR

Reply:
Sounds good to me, would make the change!

Reply:
Thanks a lot @mht-sharma !

Reply:
Done!